tests/Formatters/AbstractFormatterTests.cs

```diff
@@ -91,16 +91,25 @@ private protected async Task<SourceText> TestAsync(string testCode, string expec
                 logLevel: LogLevel.Trace,
                 saveFormattedFiles: false,
                 changesAreErrors: false,
-                filesToFormat: ImmutableHashSet.Create(document.FilePath));
+                filesToFormat: ImmutableHashSet.Create(document.FilePath),
+                reportPath: string.Empty);
 
             var filesToFormat = await GetOnlyFileToFormatAsync(solution, editorConfig);
 
             var formattedSolution = await Formatter.FormatAsync(solution, filesToFormat, formatOptions, Logger, default);
-            var formattedDocument = GetOnlyDocument(formattedSolution);
+            var formattedDocument = GetOnlyDocument(formattedSolution.Solution);
             var formattedText = await formattedDocument.GetTextAsync();
 
             Assert.Equal(expectedCode, formattedText.ToString());
 
+            if (testCode != expectedCode)
```

Comment:
`if`s in tests are a no-no, but this would require a fairly moderate refactor of this structure. 

1. Pass an additional boolean param to the `TestAsync` function, `expectingFileChanges`, which will replace this check.
2. Break this out into its own fixture, which tests the `FormattedFiles` behaviour in isolation.

Personally, I'd prefer #2, but I'll defer to you. Any particular preference?

Reply:
I would also prefer number 2, thanks!